source/common/grpc/google_async_client_impl.h

```diff
@@ -255,7 +255,7 @@ class GoogleAsyncStreamImpl : public RawAsyncStream,
     // End-of-stream with no additional message.
     PendingMessage() = default;
 
-    const absl::optional<grpc::ByteBuffer> buf_;
+    const absl::optional<grpc::ByteBuffer> buf_{};
```

Comment:
Interesting... Why are the braced needed?

Reply:
This is actually something related to the slight behavioral difference between absl::optional and std::optional. In C++17, absl::optional forwards to std::optional. 

However, in absl::optional, it provides a user-defined default constructor while std::optional doesn't do that. The reason why a user-defined default constructor is a must is because there is a const member in the data structure, and the default constructor is deleted automatically if "any non-variant non-static data member of const-qualified type (or array thereof) with no brace-or-equal-initializer does not have a user-provided default constructor."

If you want to read more about this here is the link any non-variant non-static data member of const-qualified type (or array thereof) with no brace-or-equal-initializer does not have a user-provided default constructor,